test/Microsoft.AspNet.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -685,6 +842,19 @@ private class Model
             public NestedModel NestedModel { get; set; }
 
             public bool IsACar { get; set; }
+
+            [DataType(DataType.Date)]
+            public DateTime Date { get; set; }
```

Comment:
fyi decided to leave this alone because the `DateTimeLocal` and `Time` properties could also be named `DateTimeWithDataType` or `DateTimeOffsetWithDataType`.  got too confusing.  the properties are all named after how they're intended to be used.